ansible/modules/hashivault/hashivault_write.py

```diff
@@ -96,23 +96,19 @@ def hashivault_write(params):
     data = params.get('data')
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
-        if params.get('update'):
-            read_data = client.read(secret)
-            if read_data and 'data' in read_data:
-                write_data  = read_data['data']
-            else:
-                write_data  = {}
-            write_data.update(data)
-            returned_data = client.write((secret), **write_data)
-            if returned_data:
-                result['data'] = returned_data
-            result['msg'] = "Secret %s updated" % secret
+        read_data = client.read(secret)
```

Comment:
My concern here is now hashivault_write is going to read even for the cases where update is not set

Reply:
Yeah, considering that you could have credentials that allow write but not read, I get how you could be concerned about this... I'm not sure if `hashivault_write` is often used in such use cases
But also, I find the proposed behavior more _ansible-y_.
Are you concerned in particular with the fact that the default behavior changes, or the fact that the old behavior disappears ?

Reply:
My comment was about performance, but I probably shouldn't be concerned